src/dataflow/components/stats.tsx

```diff
@@ -0,0 +1,276 @@
+import { inject, observer } from "mobx-react";
+import { BaseComponent, IBaseProps } from "./dataflow-base";
+import * as React from "react";
+import { exportCSV } from "../utilities/export";
+import "./stats.sass";
+
+export interface StatsGroup {
+  studentName: string;
+  documents: number;
+  programDocuments: number;
+  runProgramDocuments: number;
+  runProgramDataDocuments: number;
+  runProgramRelayOnlyDocuments: number;
+  publishedProgramDocuments: number;
+}
+
+interface IProps extends IBaseProps {}
+
+interface IState {
+  loading: boolean;
+  userStats: StatsGroup[];
+}
+
+const kHeaderNames = ["Documents", "Editable Programs", "Run Programs",
+                      "Run Programs w/ Data", "Run Programs Relay-Only", "Published Programs"];
+
+@inject("stores")
+@observer
+export class StatsComponent extends BaseComponent<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      loading: true,
+      userStats: [],
+    };
+  }
+
+  public async componentDidMount() {
+    const {db} = this.stores;
+    // request the published documents
+    const publicationsPath = db.firebase.getClassPersonalPublicationsPath(db.stores.user);
+    const publicationsRef = db.firebase.ref(publicationsPath);
+    const publicationsSnapshot = await publicationsRef.once("value");
+    const publishedDocOriginIds: string[] = [];
+    if (publicationsSnapshot) {
+      const pulishedDocs = publicationsSnapshot.val();
+      Object.keys(pulishedDocs).forEach((pulishedDoc) => {
+        const pulishedDocEntry = pulishedDocs[pulishedDoc];
+        if (!publishedDocOriginIds.find(originDoc => originDoc === pulishedDocEntry.originDoc)) {
+          publishedDocOriginIds.push(pulishedDocEntry.originDoc);
+        }
+      });
+    }
+
+    // request the users
+    const usersPath = db.firebase.getUsersPath(db.stores.user);
+    const usersRef = db.firebase.ref(usersPath);
+    const usersSnapshot = await usersRef.once("value");
+    if (usersSnapshot) {
+      const users = usersSnapshot.val();
+      const userStats: StatsGroup[] = [];
+
+      Object.keys(users).forEach((userId) => {
```

Comment:
```suggestion
      each(users, (userEntry, userId) => {
```
Similar to above comment, Lodash `each` simplifies the code here and eliminates the `userEntry` line below.

Reply:
Changed to Lodash `each`.